src/containers/pages/InterventionPlan/IRS/plan/index.tsx

```diff
@@ -568,6 +591,57 @@ class IrsPlan extends React.Component<
     );
   }
 
+  /** getAllAssignments - funciton to get all assignments for the plan from OpenSRP
+   * @param {string} planId the id of the plan for which to return assignments
+   * @param {Organization[]} organizations an array of Organizations by which to query assignments
+   * @param {(assignments:Assignment[])=>any} callback a function handling the resulting Assignments
+   * @returns {(assignments:Assignment[])=>any} the callback passing the Assignments as the first parameter
+   */
+  public getAllAssignments(
+    planId: string,
+    organizations: Organization[],
+    callback: (assignments: Assignment[]) => any
+  ) {
+    // define a reference of all organization ids
+    const organizationIds: string[] = organizations.map((o: Organization) => o.identifier);
+
+    // generate api calls for each organization id
+    const assignmentPromises = organizationIds.map(
+      (o: string) =>
+        new Promise((resolve, reject) => {
+          OpenSrpAssignedService.read(o, { plan: planId })
+            .then(j => resolve(j))
+            .catch(error => reject(error));
+        })
+    );
```

Comment:
Currently we are getting Assignments by organization and plan, requiring an api call for each known organization.